src/core/signals.pm6

```diff
@@ -1,13 +1,14 @@
 my enum Signal (
     |nqp::stmts(
-        ( my $res  := nqp::hash ),
+        ( my $res  := nqp::list ),
         ( my $iter := nqp::iterator(nqp::getsignals) ),
         nqp::while(
             $iter,
             nqp::stmts(
-                ( my $p := nqp::shift($iter) ),
-                nqp::bindkey($res, nqp::iterkey_s($p), nqp::abs_i(nqp::iterval($p)) )
-            )
+                ( my $p := nqp::p6bindattrinvres(nqp::create(Pair), Pair,'$!key', nqp::shift($iter)) ),
```

Comment:
FYI, looks like you're missing a space after the comma between `Pair` and `'$!key'`.